src/main/java/de/chojo/repbot/commands/Log.java

```diff
@@ -0,0 +1,131 @@
+                        .add("donated", "<user> [count]", "command.log.sub.donated")
+                        .add("message", "<message_id>", "command.log.sub.message")
+                        .build(),
+                Permission.ADMINISTRATOR);
+        this.shardManager = shardManager;
+        this.reputationData = new ReputationData(dataSource);
+    }
+
+    @Override
+    public boolean onCommand(MessageEventWrapper eventWrapper, CommandContext context) {
+        if (context.argsArray().length <= 1) return false;
+        var cmd = context.argString(0).get();
+        if ("received".equalsIgnoreCase(cmd) || "donated".equalsIgnoreCase(cmd)) {
+            var userArg = context.argString(0).get();
+            var optUser = DiscordResolver.getUser(shardManager, userArg);
+            if (optUser.isEmpty()) {
+                eventWrapper.replyErrorAndDelete(eventWrapper.localize("error.userNotFound"), 15);
+                return true;
+            }
+            if ("received".equalsIgnoreCase(cmd)) {
+                return received(eventWrapper, context.subContext(cmd), optUser.get());
+            }
+            return donated(eventWrapper, context.subContext(cmd), optUser.get());
+        }
+        if ("message".equalsIgnoreCase(cmd)) {
+            return message(eventWrapper, context.subContext(cmd));
+        }
+        return false;
+    }
+
+    private boolean message(MessageEventWrapper eventWrapper, CommandContext subContext) {
+        var optMessageId = subContext.argLong(0);
+
+        if (optMessageId.isEmpty()) {
+            eventWrapper.replyErrorAndDelete(eventWrapper.localize("error.invalidMessage"), 15);
+            return false;
+        }
+
+        var messageLog = reputationData.getMessageLog(optMessageId.get(), eventWrapper.getGuild(), 50);
+
+        var log = mapLogEntry(eventWrapper, messageLog, ReputationLogEntry::getDonorId);
+
+        var message = new LocalizedEmbedBuilder(eventWrapper)
+                .setAuthor(eventWrapper.localize("command.log.messageLog", Replacement.create("ID", optMessageId.get())))
+                .setDescription(log)
+                .build();
+        eventWrapper.reply(message).queue();
+        return true;
+    }
+
+    private void sendUserLog(MessageEventWrapper eventWrapper, User user, String title, String log) {
+        var message = new LocalizedEmbedBuilder(eventWrapper)
+                .setAuthor(eventWrapper.localize(title,
+                        Replacement.create("USER", user.getAsTag())),
+                        user.getAvatarUrl())
+                .setDescription(log)
+                .build();
+        eventWrapper.reply(message).queue();
+
+    }
+
+    private boolean donated(MessageEventWrapper eventWrapper, CommandContext context, User user) {
+        var limit = context.argInt(1).orElse(10);
+
+        var userDonatedLog = reputationData.getUserDonatedLog(user, eventWrapper.getGuild(), Math.max(5, Math.min(limit, 50)));
```

Comment:
Code probably can be further reduced by using a Supplier and a TriFunction<User, Guild, int, T>
Not 100% needed, just an idea.